apps/core/scenes/search/Search.js

```diff
@@ -80,36 +87,69 @@ class Search extends React.Component<Props> {
     this.props.setModalType(MODAL_TYPE.TRIP_TYPE);
   };
 
+  handleTripTypeSelect = (type: string) => {
+    switch (type) {
+      case 'return':
+      case 'oneWay':
+        this.props.setTripType(type);
+        break;
+      default:
+        break;
+    }
+  };
+
   handlePassengersPress = () => {
     this.props.setModalType(MODAL_TYPE.PASSENGERS);
   };
 
   render() {
-    const { tripType, adults, infants, bags } = this.props;
+    const { tripType, adults, infants, bags, layout } = this.props;
+    const desktopLayout = layout >= LAYOUT.desktop;
+
     return (
       <>
         <View style={styles.container}>
-          <View style={styles.form}>
+          <View style={[styles.form, desktopLayout && styles.formDesktop]}>
             <Illustration name="Boarding" style={styles.boardingIllustration} />
             <View style={styles.top}>
-              <TripTypeButton
-                onPress={this.handleTripTypePress}
-                icon={TRIP_TYPE[tripType].icon}
-                label={TRIP_TYPE[tripType].label}
-              />
+              {Platform.OS === 'web' ? (
+                <TripTypeSwitch
+                  style={desktopLayout && styles.flexTripTypeSwitch}
+                  optionsData={TRIP_TYPE}
+                  selectedType={tripType}
+                  onSelect={this.handleTripTypeSelect}
+                />
+              ) : (
+                <TripTypeButton
+                  onPress={this.handleTripTypePress}
+                  icon={TRIP_TYPE[tripType].icon}
+                  label={TRIP_TYPE[tripType].label}
+                />
+              )}
               <View style={styles.hSpacer} />
               <PassengersButton
                 onPress={this.handlePassengersPress}
-                passengers={adults + infants}
+                adults={adults}
+                infants={infants}
                 bags={bags}
               />
             </View>
 
-            <Placepickers />
-            <Datepickers />
+            <View style={desktopLayout && styles.inputsDesktop}>
+              <Placepickers layout={layout} />
```

Comment:
Also why is this prop-drilled down to the component, can it not import the layout there?

Reply:
Prop drilling removed.